public/style_editor.js

```diff
@@ -8410,12 +8420,71 @@ function ClickRotate() {
   console.log("ROTATE");
 }
 
+/*
+Compute delay for triggering ignition/postoff.
+Use whichever has the longest transition.
+For ignition delay, check any/all EFFECT_PREON layers.
+For POSTOFF delay, check any/all IN_TR, EFFECT_IGNITION, or EFFECT_RETRACTION layers.
+*/
 function ClickPower() {
   STATE_ON = !STATE_ON; STATE_LOCKUP=0;
   var power_button = FIND("POWER_BUTTON");
   power_button.classList.toggle("button-latched", STATE_ON ? true : false);
   console.log("POWER");
-  blade.addEffect(STATE_ON ? EFFECT_IGNITION : EFFECT_RETRACTION, Math.random() * 0.7 + 0.2);
+
+  function styleHasPreonOrPostoff(style, effectType) {
+    return style.LAYERS && style.LAYERS.some(l =>
+      l.constructor && l.constructor.name === 'TransitionEffectLClass' &&
+      l.EFFECT && l.EFFECT.getInteger &&
+      l.EFFECT.getInteger(0) === effectType
+    );
+  }
+
+  // Recursively sum transition durations
+  function getDur(n) {
+    if (n.MILLIS) return n.MILLIS.getInteger(0);
+    var sum = 0;
+    if (n.args) for (let a of n.args) sum += getDur(a);
+    return sum;
+  }
+
+  // If turning ON and the style has PREON, trigger it and delay ignition until it finishes.
+  if (STATE_ON) {
+    if (styleHasPreonOrPostoff(current_style, EFFECT_PREON)) {
+      blade.addEffect(EFFECT_PREON, 0.0);
+      // Use the longest PREON transition to delay ignition.
```

Comment:
This feels a little.. backwards... since the length of the preon doesn't come from the style in ProffieOS.
Maybe we just use the wavlen setting for this?  Or maybe do it only if you have a font loaded?

Reply:
Ah. good point. Preon duration is based on sound file length in the real world.
So yeah, then this needs to be reevaluated as to how it should work, and that in itself depends on if you want to do the integrated sound thing.
I started making a list of all the possible combinations that could happen, like 
style has no preon, 
has preon but no WavLen<> value, 
has preon but the single globally set WavLen time isn't accurate, so it looks wrong, 
has preon but .......
and then I stopped and decided that in ordere to have linked preon and postoff work right, it kind of really _needs_ to use an accurate WavLen system. So...